### DISCUSSION TITLE
Getting a "diff" of two different OSCAP releases?

### DISCUSSION THREAD

**Discussion by PeterWhittaker** [ComplianceAsCode/content#10509] [Q&A]
Apologies if this is the wrong forum, I've been trying to determine the best place to pose this question, but haven't had much luck (the oscap anaconda add-on mailing list, e.g., has very little traffic and the last few Qs remain unanswered).

I am trying to get a "diff" of the commands that are executed as part of applying different releases of the `oscap-anaconda-addon`, specifically the STIGs applied in `RHEL 7.9` and in `AlmaLinux 9.1`. Context: We have a security appliance that was originally based on RedHat 7.9 and are migrating it to AlmaLinux 9.1. There has been a lot of "empirical discovery" so far as we run into configuration differences between the two platforms.

Most of these we can resolve fairly readily, but the OSCAP differences are opaque, given that they are ultimately based on differences in the underlying `openscap` packages, which are complex.

Ideally, I'd like to know what remediation commands are run by the add-on for each release. I'd be happy with some direction that would allow me to write a program that parses different versions of `openscap` using an appropriate "key" and returns the list of mitigation commands, which I could then compare.

Any and all guidance will be warmly received! Thank you!

**Comment by matejak** [comment]
You are probably looking in a wrong direction, the `openscap` pacakage just contains the scanner, and the scanner attempts to do what the content tells it. So you want to look at differences between the 7.9 STIG profile, and the 9.1 one - the scanner runs remediations for rules that the profile consists of.

**Reply by PeterWhittaker** [reply]
@matejak Let me rephrase: I want exactly what you imply, a diff between the two STIG profiles, specifically the new commands run by the scanner. As far as I can tell, the only way to get this is to write code to generate it. Am I on the right track, or am I missing something?

To put it another way, in some ways my ideal solution would be an equivalent of `make -n`, that is, an invocation of openscap that returns the set of commands that would be run, without running anything. If I could generate that from the STIGs, then I could `sort` and `diff` to my heart's content. Thoughts?
